flagsmith-core.ts

```diff
@@ -167,7 +167,17 @@ const Flagsmith = class {
                     console.error(e)
                 }
             }
-
+            if(this.sentry) {
+                const flagsIntegration =
+                    this.sentry.getClient()?.getIntegrationByName(
+                        "FeatureFlags",
+                    );
+                if (flagsIntegration) {
+                    Object.keys(this.flags).map((key) => {
+                        flagsIntegration.addFeatureFlag(key, this.hasFeature(key))
```

Comment:
From my understanding of how the Sentry [integration works](https://docs.sentry.io/platforms/javascript/configuration/integrations/featureflags/), `addFeatureFlag` is expected to add a certain flag _when evaluated_ to the context of an error.

This patch might add **all feature flags** to Sentry errors, rather than only the relevant ones being evaluated and used in the context of an exception.

Other SDKs handle that by attaching a callback to evaluation hooks, or monkeypatching the client to call `add_feature_flag`. See [example](https://github.com/getsentry/sentry-python/blob/ab2e3f08b600b22a95c3313eddd66f733e2d133c/sentry_sdk/integrations/unleash.py#L29).

Reply:
Solved 🚀  - it'll only send over evaluated flags